examples/zero_inflated_poisson.py

```diff
@@ -0,0 +1,171 @@
+# Copyright Contributors to the Pyro project.
+# SPDX-License-Identifier: Apache-2.0
+
+"""
+Example: Zero-Inflated Poisson regression model
+================================================
+
+In this example, we model and predict how many fish visitor groups to a state park catch.
```

Comment:
nit:

In this example, we model and predict how many fish are caught by visitors to a state park.
Many groups of visitors catch zero fish, either because they did not fish at all or because they were unlucky. 
We would like to explicitly this bimodal behavior (zero versus non-zero) and ascertain which variables contribute to each behavior.

We answer this question by fitting a zero-inflated poisson regression model. We use MAP,
VI and MCMC as estimation methods. Finally, from the MCMC samples, we identify the variables that
contribute to the zero and non-zero components of the zero-inflated poisson likelihood.